### PR TITLE
feat(repo): add Dora refresh helper and guidance

### PR DESCRIPTION
**Pull request by jrschumacher**
## Summary
- add repo guidance telling agents to run the Dora refresh helper proactively before broad code exploration
- add repo-tracked Dora helper scripts to create local config, build a merged Go workspace SCIP index, and refresh the local Dora database
- ignore local `.dora/` artifacts and include the Dora Go workspace improvement plan

## Testing
- `bash -n scripts/dora-index.sh scripts/dora-ensure.sh`
- `cd tools/dora-merge-scip && GOWORK=off golangci-lint run ./...`
- `cd tools/dora-merge-scip && GOWORK=off go test ./...`
- `bash scripts/dora-ensure.sh`

### CODE REVIEW

File: dora-go-improvement-plan.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+- it avoids indexing empty/test-only package directories as primary packages,
+- it gives Dora stable import paths to pass to `scip-go`,
+- and it avoids some of the edge cases we saw when treating `./...` as a monolithic pattern from the wrong root.
+
+### Step 4: run `scip-go` per module
+
+For each discovered module:
+
+```bash
+scip-go index <import-paths...> --module-root <module-dir> --output <fragment>
```
[the remark this answers is not in the document]
Comment (jrschumacher) [reply dora-go-improvement-plan.md]: Thanks — I tightened the wording in the plan to use `package-patterns` here. I kept the positional arguments because the current `scip-go index` CLI does support package patterns as positional args alongside `--module-root` (per `scip-go index --help`), so the original command shape is still valid.

File: dora-go-improvement-plan.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+  },
+  "scip": {
+    "merged": ".dora/index.scip"
+  }
+}
+```
+
+Behavior:
+
+- Dora writes one fragment per module into `.dora/scip/`.
```
[the remark this answers is not in the document]
Comment (jrschumacher) [reply dora-go-improvement-plan.md]: Good catch — I updated the plan to call out unique fragment names derived from module paths.

File: dora-go-improvement-plan.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+Parse `go.work` and capture all `use (...)` entries.
+
+### Step 3: discover indexable packages per module
+
+For each module directory, use Go tooling to identify real packages that should be indexed.
+
+A good default is something equivalent to:
+
+```bash
+go list -e -f '{{if or .GoFiles .CgoFiles}}{{.ImportPath}}{{end}}' ./...
```
[the remark this answers is not in the document]
Comment (jrschumacher) [reply dora-go-improvement-plan.md]: Good call — I updated the example to filter empty lines from the `go list` output.
